build-scripts/plashet.py

```diff
@@ -211,7 +212,25 @@ def is_signed(config, nvr):
     :return: Returns whether the specified nvr is signed with the signing key id. An exception
     will be raise if the nvr can't be found at all in the brew root (i.e. unsigned can't be found).
     """
-    return os.path.isdir(get_brewroot_arch_base_path(config, nvr, True))
+    signed_base = get_brewroot_arch_base_path(config, nvr, True)
+    unsigned_base = get_brewroot_arch_base_path(config, nvr, False)
+
+    if os.path.isdir(signed_base):
+        # The signed directory exists, but we also want to make sure that the RPM counts match
+        # the unsigned directories. This eliminates a potential race condition between a nvr
+        # being signed and the time it takes to populate the brewroot directories.
+
+        signed_rpm_count = len(glob.glob(f'{signed_base}/**/*.rpm', True))
+        # Note the structure brewroot has signed under the unsigned directory, so the next
+        # glob will find both signed and unsigned.
+        all_rpm_count = len(glob.glob(f'{unsigned_base}/**/*.rpm', True))
+
+        if all_rpm_count != (signed_rpm_count * 2):
+            logger.info('fFound incomplete signed rpm directory for {nvr}; brewroot may still be being built.')
```

Comment:
ITYM
```suggestion
            logger.info(f'Found incomplete signed rpm directory for {nvr}; brewroot may still be being built.')
```

otherwise looks good... nice solution.